pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/QueryGenerator.java

```diff
@@ -723,25 +771,53 @@ public Query generateQuery() {
         groupColumns.add(pickRandom(_singleValueColumnNames));
       }
 
+      ArrayList<String> arrayOfAggregationColumnsAndFunctions = new ArrayList<>(aggregationColumnsAndFunctions);
+
+      //Generate a HAVING clause for group by query
+      String havingClause = createHavingClause(arrayOfAggregationColumnsAndFunctions);
+
       // Generate a result limit of at most MAX_RESULT_LIMIT.
       TopQueryFragment top = new TopQueryFragment(RANDOM.nextInt(MAX_RESULT_LIMIT + 1));
 
-      return new AggregationQuery(new ArrayList<>(aggregationColumnsAndFunctions), predicate, groupColumns, top);
+      return new AggregationQuery(arrayOfAggregationColumnsAndFunctions, predicate, groupColumns, top, havingClause);
     }
-  }
 
-  /**
-   * Predicate generator interface with capability of generating a predicate query fragment on a column.
-   */
-  private interface PredicateGenerator {
+    private String createHavingClause( ArrayList<String> arrayOfAggregationColumnsAndFunctions) {
+      String havingClause = new String();
```

Comment:
Can this generate queries with aggregation functions that are not in the select list? Have you tested this overnight (eg. with 100k+ queries)?